app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx

```diff
@@ -3,26 +3,37 @@ import React from 'react';
 import { QueryConstraint, QueryVerificationStatus } from '../types';
 
 type AllStatuses = QueryConstraint['status'] | QueryVerificationStatus;
+export const ACTION_DISPLAY_NAMES = ['passed', 'overridden', 'pending', 'failed'] as const;
+export type ActionDisplayName = typeof ACTION_DISPLAY_NAMES[number];
 
-export const actionStatusUtils: {
-  [key in AllStatuses]: { color?: string; icon: string; displayName: string };
-} = {
+const DEFAULT_ICON = 'far fa-hourglass';
+
+type ActionStatusUtils = {
+  [key in AllStatuses]: { color?: string; icon: string; displayName: ActionDisplayName };
+};
+
+// This ensures that we cover all of the options
+const actionStatusUtilsInternal: ActionStatusUtils = {
   FAIL: { color: 'var(--color-status-error)', icon: 'fas fa-times', displayName: 'failed' },
   FORCE_PASS: { color: 'var(--color-status-success)', icon: 'fas fa-check', displayName: 'overridden' },
   PASS: { color: 'var(--color-status-success)', icon: 'fas fa-check', displayName: 'passed' },
-  PENDING: { icon: 'far fa-hourglass', displayName: 'pending' },
-  NOT_EVALUATED: { icon: 'far fa-hourglass', displayName: 'pending' },
+  PENDING: { icon: DEFAULT_ICON, displayName: 'pending' },
+  NOT_EVALUATED: { icon: DEFAULT_ICON, displayName: 'pending' },
+  BLOCKED: { icon: DEFAULT_ICON, displayName: 'pending' },
 };
 
+// This forces the consumer to verify that the value exists (important in the case of a breaking change on the backend)
+export const actionStatusUtils: Partial<ActionStatusUtils> = actionStatusUtilsInternal;
+
```

Comment:
Can you explain this a bit more? I don't understand what the reassignment does here.

Reply:
You can see below where I use the object the return type of this `actionStatusUtils[status]` can be undefined and I have to use `?`. By adding this I'm making sure that adding new options to the backend won't cause the frontend to crash

Reply:
I don't understand why you can't use `actionStatusUtilsInternal[status]?.icon` If a new status is added to the backend, then `actionStatusUtilsInternal[status]` would also be undefined, and the optional chaining would serve the same purpose.

Reply:
That's exactly what I'm doing, the purpose of line 26 is to force you to do that :)

Reply:
I am not seeing a difference between `actionStatusUtilsInternal[status]?.icon` and `actionStatusUtils[status]?.icon`, so I don't think that line 26 is necessary.

Reply:
The difference is:
- `actionStatusUtilsInternal[status].icon` is valid and won't raise an error
- `actionStatusUtils[status].icon` this is not valid and therefore forces the consume to handle the undefined case properly

Reply:
But aren't they the same thing? This line just says `actionStatusUtils = actionStatusUtilsInternal`